internal/istiostatus/istio_status_checker.go

```diff
@@ -10,18 +10,17 @@ import (
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 )
 
-type istioStatusChecker struct {
-	client client.Reader
+type Checker struct {
+	Client client.Reader
```

Comment:
Could we maybe internalize it? (client lower-case)

Reply:
Ah, for that you would need a ctor function (`NewChecker`)

Reply:
fixed